File: android/app/src/main/java/com/dontworry/app/domain/search/ExcerptBuilder.kt
package com.dontworry.app.domain.search

object ExcerptBuilder {
    fun fromText(text: String?, maxLength: Int = 180): String {
        if (text.isNullOrBlank()) return "No content preview available."
        val clean = text.replace("\\s+".toRegex(), " ").trim()
        return if (clean.length <= maxLength) clean else clean.take(maxLength).trimEnd() + "..."
    }
}

File: ios/DontWorryiOS/Domain/ExcerptBuilder.swift
import Foundation

enum ExcerptBuilder {
    static func fromText(_ text: String?, maxLength: Int = 180) -> String {
        guard let text, !text.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
            return "No content preview available."
        }

        let clean = text
            .replacingOccurrences(of: #"\s+"#, with: " ", options: .regularExpression)
            .trimmingCharacters(in: .whitespacesAndNewlines)

        if clean.count <= maxLength {
            return clean
        }

        let index = clean.index(clean.startIndex, offsetBy: maxLength)
        return clean[..<index].trimmingCharacters(in: .whitespacesAndNewlines) + "..."
    }
}
